### ISSUE TITLE
DependencyTelemetry.TryGetOperationDetail isn't as helpful as it could be

### ISSUE THREAD

**Issue by IanKemp** [microsoft/ApplicationInsights-dotnet-server#1260]
See #897.

1. There is no indication as to what values can be passed as the key.
2. There is no type-safety in the values returned.

tl;dr there's no way for anyone to know that they can call

```csharp
var request = myDependencyTelemetryInstance.TryGetOperationDetail("HttpRequest") as HttpRequestMessage;
```

The only reason I know to do the above is because I trawled through issues and PRs in this repo. Most people probably won't be as patient, and they shouldn't need to be.

#### Suggestion

1. Publicly expose relevant [`RemoteDependencyConstants`](https://github.com/microsoft/ApplicationInsights-dotnet-server/blob/master/Src/DependencyCollector/Shared/Implementation/RemoteDependencyConstants.cs#L21) keys.
2. Add specific overloads on `DependencyTelemetry` that reflect what `HttpCoreDiagnosticSourceListener` and `HttpProcessing` are storing:

   `HttpRequestMessage TryGetHttpRequestOperationDetail();`
   `HttpResponseMessage TryGetHttpResponseOperationDetail();`
   `WebHeaderCollection TryGetHttpResponseHeadersOperationDetail();`
3. Actually publicise the existence of these predefined keys and values in the [official docs for `DependencyTelemetry.TryGetOperationDetail`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.applicationinsights.datacontracts.dependencytelemetry.trygetoperationdetail?view=azure-dotnet)

**Comment by cijothomas** [comment]
@IanKemp you are right, this feature is not publicly documented.  It's a good idea to public expose the keys. Tagging as an enhancement.

**Comment by cijothomas** [comment]
Need to evaluate if similar concept can be extended to Requesttelemetry as well. We have quite large number of customers asking for mechanism to read request/response body. While it can't be done by SDK by default (privacy etc.), if we populate the request/response objects in this way, users can access it in initializer and add to telemetry as desired.
